lib/algolia/responses/dictionary_response.rb

```diff
@@ -0,0 +1,37 @@
+module Algolia
+  class DictionaryResponse < BaseResponse
+    include CallType
+
+    attr_reader :raw_response
+
+    # @param client [Search::Client] Algolia Search Client used for verification
+    # @param response [Hash] Raw response from the client
+    #
+    def initialize(client, response)
+      @client       = client
+      @raw_response = response
+      @done         = false
+    end
+
+    # Wait for the task to complete
+    #
+    # @param opts [Hash] contains extra parameters to send with your query
+    #
+    def wait(_opts = {})
+      retries_count = 1
+
+      until @done
+        res               = @client.custom_request({}, path_encode('/1/task/%s', @raw_response[:taskID]), :GET, READ)
+        status            = get_option(res, 'status')
+        if status == 'published'
+          @done = true
+        end
+        retries_count    += 1
+        time_before_retry = retries_count * Defaults::WAIT_TASK_DEFAULT_TIME_BEFORE_RETRY
```

Comment:
This increases the delay with each loop that we do: is that intended? If so, why don't we keep the time between retries the same?

 E.g. if the first loop we wait 1 second, we wait 5 seconds at the fifth loop. Why not keep it 1 second all the time?

Reply:
we wait a bit longer in case there's latency

Reply:
The latency is handled by the `@client`, right? Basically, with each loop we're waiting longer and longer to check the status of a task, while the chance of it being done is bigger the longer we wait. So I would actually expect that the time between requests stays the same or even reduces, if that makes sense. WDYT?